### PR TITLE
concours: online mode for concours

### PR DESCRIPTION
**Pull request by lportemo**
TODO:

- [X] Settings
- [X] Urls
- [X] Adapt `login.html' template
- [x] Add logout button to the navbar when online mode is enabled
- [x] Test it !
- [x] Add a API Key page when online mode is enabled

### CODE REVIEW

File: prologin/concours/stechec/views.py [hunk 1]
```diff
@@ -782,3 +785,13 @@ class ReportBugList(RedmineIssueListView):
         ('tracker_id', '=', 1),
         ('author_id', '=', 'me'),
     ]
+
+
+class LoginView(django.contrib.auth.views.LoginView):
```
Comment (seirl) [inline prologin/concours/stechec/views.py:790]: I don't think you need that, we already make the settings available to all the templates here: https://github.com/prologin/sadm/blob/82aa204b659b804fe6fc46164b9d41978c3331b8/prologin/concours/stechec/context_processors.py

You should just be able to use `settings` directly in the view.

Comment (lportemo) [reply prologin/concours/stechec/views.py:790]: thanks, I'll change that :)

**Comment by lportemo** [comment]
To help reviewers understand the details of this PR :
I am using the `password` field of the `django.contrib.auth.get_user_model()` to store user's API key.

Few reasons for this :

- online mode means that you are doing OAuth2 with external site, so you don't have the password set in `password` field in concours db
- Django Rest Framework allows Basic Authentication with your username and password. Concours' users in online mode must give their username and api key (as password) to be authenticated/authorized in the API
- Why store it in cleartext ? Because it's easier for organizers to just get all the API keys of the registered users to feed them in another first party software that relies on concours' REST API

File: prologin/concours/webide/rules.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+from . import models
+from django.core.exceptions import ObjectDoesNotExist
+from django.contrib.auth import get_user_model
+
+import requests
+
+class WorkspaceVoucher:
```
Comment (audebert) [inline prologin/concours/webide/rules.py:7]: Add documentation, it's not trivial what are the dependencies for this class, and what it does.

File: requirements.txt [hunk 3]
```diff
@@ -23,4 +23,9 @@ tornado==4.5.1
 wiki==0.5
 
 # From prologin
+git+https://github.com/prologin/django-proloauth-client
 -e git://github.com/prologin/camisole.git#egg=camisole
+
+# finale2020 please delete
```
[the remark this answers is not in the document]
Comment (seirl) [reply requirements.txt:29]: We should put that in extra_requires (and do away with this requirements file).

**Comment by juli0z** [comment]
Migrated to https://gitlab.com/prologin/tech/infra/sadm/-/merge_requests/212

**Pull request closed without merging**